packages/app/lib/src/features/home/presentation/home/post.dart

```diff
@@ -225,30 +225,29 @@ class _PostInteractables extends HookConsumerWidget {
     final userId = ref.watch(idProvider);
     final username = ref.watch(usernameProvider);
 
-    // maybe update to use a provider instead
+    // Create a copy of the likes list
     final currentLikes = useState(
       List<String>.from(post.likes),
     );
+
     //Like button logic
-    final ValueNotifier<Icon> thumbsIcon;
-    if (currentLikes.value.contains(userId)) {
-      thumbsIcon = useState(const Icon(Icons.thumb_up_sharp));
-    } else {
-      thumbsIcon = useState(const Icon(Icons.thumb_up_outlined));
-    }
-    // Create a copy of the likes list
+    final thumbsIcon = useState(
+      currentLikes.value.contains(userId)
+          ? Icons.thumb_up_sharp
+          : Icons.thumb_up_outlined,
+    );
```

Comment:
Why are we even using a useState? Can't we just inline this?

Reply:
I wrote the thumbIcon code like 2 months ago. I remember I originally didn't use useState() and there was some issues. Feel free to try a different method though.

Reply:
I realized that this caches it. It'd be simpler not use it, but it's probably more performant this way.